### DISCUSSION TITLE
Where are config files generated/kept on linux executable?

### DISCUSSION THREAD

**Discussion by Swizzler121** [PhilipK/BoilR#212] [Q&A] [answered]
So I was kinda bummed to find out that the flatpak version doesn't support detecting flatpak games because of a limitation of flatpak, so I looked into the executable, and was surprised it doesn't really have any install process other than flagging it executable. Makes me wonder why can't you just use that on the steam deck?

Does it store it's configs in userspace? I ran it for a bit and it didn't generate any files in the directory I stuck it in, so I'm not sure what it generates and how it stores settings, but if everything lives under /home, theoretically that should work fine on deck, right? That's the whole reason flatpaks  and appimages play nice with the Steam Deck, is they confine everything to the home directory so it doesn't mess anything up when the system is re-imaged.

Is there info anywhere about what files BoilR generates, and where it stores them, or even a config file where that's all kept?

**Comment by PhilipK** [comment, marked as answer]
Can the native version be used on the steam deck (or another immutable OS) with no problem? Yes, the only advantage to the flatpak is the auto update feature. Besides that the native version is smaller, runs a bit faster and has more launchers it can find.
The reason for making the flatpak was that a lot of people where asking for it, for the ease of use/install/update.
But I originally picked Rust as the programming language because it can make very small self-contained executables, that do not need anything to be installed.

Where is the configuration file then?
Well it depends on the setup and the OS.
This file defines all the different options there are:
https://github.com/PhilipK/BoilR/blob/main/src/config.rs

For most linux users , the default location used will be ~/.config/boilr/ , on the steam deck that would be /home/deck/.config/boilr/ .
If you define a environment variable XDG_CONFIG_HOME (which flatpak does when it runs a program) it will use that.

On windows it would be in %APPDATA%/boilr/ if it has access, and next to the executable if it does not.

So, everything lives in the home drive by default and can survive a OS update, and it has no OS dependencies, so even if some library gets update the executable does not care (this is for both Windows and Linux).

**Reply by Swizzler121** [reply]
Awesome! thanks for the clarification, I had just found the config file while I was actually playing around with xdg-ninja and sprucing up my home dir, so I was coming back to let you know, thanks for the quick reply!
